src/server/project.ts

```diff
@@ -107,6 +107,7 @@ namespace ts.server {
         private rootFilesMap: FileMap<ScriptInfo> = createFileMap<ScriptInfo>();
         private program: ts.Program;
         private externalFiles: SortedReadonlyArray<string>;
+        private missingFilesMap: FileMap<FileWatcher> = createFileMap<FileWatcher>();

```

Comment:
I don't understand when we use `Path` and `FileMap`.  Do you have some time to explain?